### ISSUE TITLE
k8tz sidecar container: pulling from private registry fails

### ISSUE THREAD

**Issue by hrbu** [k8tz/k8tz#49]
In our project we pull images from a secured internal container-registry. Therefore we provide an imagePullSecret. 
![grafik](https://user-images.githubusercontent.com/10865951/205845452-2634da4a-d200-4591-9550-a865eff27c60.png)

Pulling the image for the k8tz controller deployment works as expected.

**Problem**: Pulling the image for sidecar container usage fails. 

Seems like the imagePullSecret isn't considered in that case.

**Comment by yonatankahana** [comment]
hi, can you provide me the full error message?
and show the full repository value that cutted in the image?
a print of `kubectl get pod -oyaml -n <namespace> <podName>` can be useful as well.

**Comment by hrbu** [comment]
Providing full error messages is not that easy ... the problem exists on multinode k8s environments only. So reproducing on my local k3d setup is not possible. For more detailed error messages I've to roll-back a pre-production environment in that error state. Would like to avoid this if possible. During investigation of the deployment incident I've seen error messages containing "access denied" when pulling the image for sidecar-container usage. Within the related pod deployment description I've missed the imagePullSecrets section for the k8tz sidecar-container.

The following screenshot is what I can provide without much effort:

![grafik](https://user-images.githubusercontent.com/10865951/205849270-3d585b7b-8c77-4ec4-99d0-027788ebc24c.png)

**Comment by yonatankahana** [comment]
you are right, imagePullSecrets not propagated from controller to sidecars and it will cause this problem. I need to fix it. ill update you soon
